### PR TITLE
Updates so user can deploy their own personal token

### PR DESCRIPTION
**Pull request by mds1**
Main changes:
- `factoryAddress` is now set to the Rinkeby pToken factory address
- Updates to `deployAndSaveToken()` to properly handle errors and deployment
- Personal token contract can now be deployed from dashboard

@coopahtroopanew No real UX work was done here, the big one left to do related to this PR is `// TODO: Show loading while deploying`. Did add a call to `indicateMetamaskPopup()` to show the pointer to the browser extension when a tx needs to be confirmed, but I haven't tested how that works with other wallets (hopefully it does nothing, but I'm not sure)

### CODE REVIEW

File: app/assets/v2/js/board.js [hunk 1]
```diff
@@ -318,20 +322,34 @@ Vue.mixin({
...
+        _alert('Please connect a wallet', 'error');
+      }
       [user] = await web3.eth.getAccounts();
-      // TODO: This is a deterministic localhost address. Should be an env variable for rinkeby/mainnet.
-      const factoryAddress = '0x7bE324A085389c82202BEb90D979d097C5b3f2E8';
-      const mockDaiAddress = '0x6b67DD1542ef11153141037734D21E7Cbd7D9817';
+
+      // We currently have DAI addresses hardcoded, so right now pTokens only support
+      // being priced in DAI
+      let tokenAddress;
```
Comment (proofoftom) [inline app/assets/v2/js/board.js]: I feel like simply calling this `tokenAddress` may be confused with `pTokenAddress`. At risk of being overly explicit, perhaps we call this `purchaseTokenAddress`?

File: app/assets/v2/js/pages/profile_tokens.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
-const factoryAddress = '0x7bE324A085389c82202BEb90D979d097C5b3f2E8';
-const DAI = '0x6b67DD1542ef11153141037734D21E7Cbd7D9817';
+// Personal token constants
+// Note that this address is also duplicated in board.js
+const factoryAddress = '0x80D50970599E33d0D5D436A649C25b729666A015';
+const defaultTokenName = 'DAI';
```
Comment (proofoftom) [inline app/assets/v2/js/pages/profile_tokens.js]: Same as above, perhaps we call this `purchaseTokenName`

File: app/assets/v2/js/pages/profile_tokens.js [hunk 3]
```diff
@@ -70,6 +72,8 @@ async function buyPToken(tokenAmount) {
     pTokenAddress // TODO: this needs to be derived from the profile page
   );
 
+  // TODO add token approval step
```
Comment (proofoftom) [inline app/assets/v2/js/pages/profile_tokens.js:75]: Will these TODOs be addressed in a separate PR?

Comment (mds1) [reply app/assets/v2/js/pages/profile_tokens.js:75]: Yea this TODO is for purchasing ptokens, whereas this PR is only for deploying them. So that will be addressed separately

File: app/assets/v2/js/board.js (hunk 1 above)
Comment (mds1) [reply app/assets/v2/js/board.js]: Updated in b4ec2ce

File: app/assets/v2/js/pages/profile_tokens.js (hunk 2 above)
Comment (mds1) [reply app/assets/v2/js/pages/profile_tokens.js]: Updated in b4ec2ce
